### PR TITLE
Replace ring with sha2.

### PR DESCRIPTION
**Pull request by jdm**
ring doesn't build on some platforms that Servo builds for, so let's use a crate with a more restricted feature set.

### CODE REVIEW

**Comment by Darkspirit** [comment]
Please wait for a response at https://github.com/briansmith/ring/issues/960. aarch64 windows needs to be added to https://github.com/briansmith/ring/blob/ddfa683266e8ce41c06feb9f46503dd4e1d47ab1/build.rs#L230

**Comment by jdm** [comment]
One thing I like about this change is that we're only pulling in a crate that does digests, rather than everything else that ring does.

**Comment by Darkspirit** [comment]
The point is, WebRender recently [removed sha2](https://github.com/servo/webrender/commit/c02872085918bab55e4feeec01390f856aa67eb1#diff-d2ede298d9a75c849170d6ef285eda1e), that would remove it from Servo's Cargo.lock, too.

It would be good to use to same high-performance library for CSP and [subresource integrity](https://github.com/servo/servo/blob/master/components/net/subresource_integrity.rs) (currently using OpenSSL - which ring is a polished subset of).

In Servo, _ring_ would unblock further enhancements like trust-dns (DNS-over-HTTPS, Encrypted SNI, DANE instead of the obsolete HPKP) and rustls ([hyper's HTTP3](https://github.com/djc/quinn/blob/47f58b446856b9cbeab28fd2496294c7d0bc085a/quinn-proto/Cargo.toml#L32), [Certificate Transparency](https://crates.io/crates/sct), using the strict [Mozilla CA root crate](https://crates.io/crates/webpki-roots) instead of the current CA roots file which doesn't/can't contain any of Mozilla's important restrictions).

**Comment by jdm** [comment]
Webrender removed sha2 because they didn't need any special hashing algorithms. The CSP implementation does, so that's not a compelling argument against sha2.

Servo uses openssl for the SRI hashing because openssl provides the necessary algorithms. That doesn't seem like a compelling reason to rely on openssl in the CSP crate, though. We could easily switch that code to use the sha2 crate instead.

Whether the CSP crate includes ring or not has no bearing on whether Servo experiments with network security enhancements like you describe.

In short, I don't see a good reason to hold out for using ring in this code.

bors r=notriddle
